### PR TITLE
Baselining Enum4Linux

### PR DESCRIPTION
**Pull request by rng4302**
Baselined Enum4Linux tool

### CODE REVIEW

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @rng4302 

This is not a full review - it is just a couple of suggestions to help you get past the CI checks. 

Kind regards

Michael

**Comment by rng4302** [comment]
Made the changes as suggested. And the check are also passing. Please inform me if any further changes needs to be done

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { UserGuide } from "../UserGuide/UserGuide";
-import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { RenderComponent } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import React from "react";
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:11]: Not required. Can remove. 
```suggestion

```

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 2]
```diff
@@ -1,42 +1,51 @@
...
+ * The Enum4Linux component.
+ */
 const Enum4Linux = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
     const [pid, setPid] = useState("");
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false);
     const [allowSave, setAllowSave] = useState(false);
     const [hasSaved, setHasSaved] = useState(false);
+    const [opened, setOpened] = useState(false);
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:34]: Should only be open if the command is not available.
```suggestion
    const [opened, setOpened] = useState(!isCommandAvailable);
```

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 3]
```diff
@@ -1,42 +1,51 @@
...
+    const [loadingModal, setLoadingModal] = useState(true);
+    const [customMode, setCustomMode] = useState(false);
+
+    const dependencies = ["enum4linux"];
+    const title = "Enum4Linux";
+    const description = "Enum4linux is a tool used for enumerating information from Windows and Samba systems.";
+    const steps =
+        "Step 1: Enter the Ip address of the target.\n" +
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:42]: IP
```suggestion
        "Step 1: Enter the IP address of the target.\n" +
```

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 4]
```diff
@@ -1,42 +1,51 @@
...
+        "Step 2: Type in the option you want to do enumeration.\n" +
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:43]: More information required. What options are available? Examples are userlist (-U), machine list (-M), etc.

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 5]
```diff
@@ -1,42 +1,51 @@
...
+        "Step 3: Enter any additional parameters you want to add.";
+    const sourceLink = ""; // Link to the source code (or Kali Tools).
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:45]: Enter the link to the Kali page about Enum4Linux.
```suggestion
    const sourceLink = "https://www.kali.org/tools/enum4linux/"; // Link to the source code (or Kali Tools).
```

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 6]
```diff
@@ -1,42 +1,51 @@
...
+    const [loadingModal, setLoadingModal] = useState(true);
+    const [customMode, setCustomMode] = useState(false);
+
+    const dependencies = ["enum4linux"];
+    const title = "Enum4Linux";
+    const description = "Enum4linux is a tool used for enumerating information from Windows and Samba systems.";
+    const steps =
+        "Step 1: Enter the Ip address of the target.\n" +
+        "Step 2: Type in the option you want to do enumeration.\n" +
+        "Step 3: Enter any additional parameters you want to add.";
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:44]: Need to add Step 4 about clicking the button. 
```suggestion
        "Step 3: Enter any additional parameters you want to add." + ;
        "Step 4: Click the Start " + title + " button and view the output block to see the results. ";
```

File: src/components/Enum4Linux/Enum4Linux.tsx [hunk 7]
```diff
@@ -46,109 +55,112 @@ const Enum4Linux = () => {
...
+                    <TextInput label="Option" required {...form.getInputProps("argumentMain")} />
+                    <TextInput label="Parameters" {...form.getInputProps("paramMain")} />
+                    {customMode && (
+                        <>
+                            <TextInput label="Additional Options" {...form.getInputProps("argumentAlt")} />
+                            <TextInput label="Additional Parameters" {...form.getInputProps("paramAlt")} />
+                        </>
+                    )}
+                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
+                    <Button type="submit">Scan</Button>
```
Comment (michaeljpigott) [inline src/components/Enum4Linux/Enum4Linux.tsx:159]: The convention is "Start <title of tool>"
```suggestion
                    <Button type="submit">Start { title }</Button>
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @rng4302 

This is looking pretty good. I did notice that the tool does not appear to be working and I am not sure why:

image

In the terminal, it looks like this: 

image

Do not stress if you cannot fix it because the main purpose of this task is to baseline the tool. 

I have requested some minor changes. 

Kind regards 

Michael

**Comment by michaeljpigott** [comment]
@rng4302 Changed the label to "Awaiting Assignee Update" because I am waiting for you to act on the changes requested. 

Kind regards

Michael

**Review by michaeljpigott — APPROVED** [review]
Hi @rng4302 

Thank you for making those changes. I have tested with the parameters that you advised and the tool works. I approve this pull request. 

Kind regards

Michael

**Pull request merged**